src/components/form/ReportTableSearchInput.js

```diff
@@ -0,0 +1,174 @@
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import React, { useState, useRef } from 'react'
+import i18n from '@dhis2/d2-i18n'
+import Paper from '@material-ui/core/Paper'
+import MenuItem from '@material-ui/core/MenuItem'
+import Popper from '@material-ui/core/Popper'
+import CircularProgress from '@material-ui/core/CircularProgress'
+import red from '@material-ui/core/colors/red'
+import { Input } from './Input'
+import { formInput, formInputMeta } from '../../utils/react/propTypes'
+import {
+    loadFilteredStandardReportTables,
+    clearSearch,
+    MIN_CHAR_LENGTH,
+} from '../../redux/actions/standardReportTables'
+
+const centeredStyle = {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    paddingTop: 16,
+    paddingBottom: 16,
+}
+
+const errorStyle = {
+    ...centeredStyle,
+    color: red[500],
+}
+
+const infoStyle = {
+    ...centeredStyle,
+    color: '#494949,',
+    fontStyle: 'italic',
+}
+
+const getListContent = ({ onClick, loading, error, collection, noMatches }) => {
+    if (loading) {
+        return (
+            <div style={centeredStyle}>
+                <CircularProgress />
+            </div>
+        )
+    } else if (error) {
+        return <div style={errorStyle}>{error}</div>
+    } else if (noMatches) {
+        return <div style={infoStyle}>{i18n.t('No matches found')}</div>
+    } else if (collection.length > 0) {
+        return collection.map(item => {
+            return (
+                <MenuItem
+                    key={item.value}
+                    onClick={() => onClick(item.value, item.label)}
+                >
+                    {item.label}
+                </MenuItem>
+            )
+        })
+    } else {
+        return null
+    }
+}
+
+export const ReportTableSearchInputUI = props => {
+    const inputRef = useRef()
+    const [isSearchMode, setSearchMode] = useState(false)
+    const [reportTableName, setReportTableName] = useState('')
+    const showResults =
+        isSearchMode && props.searchTerm.length >= MIN_CHAR_LENGTH
+    const showHint = isSearchMode && props.searchTerm.length < MIN_CHAR_LENGTH
+    const onFocus = () => {
+        if (!isSearchMode) {
+            setSearchMode(true)
+        }
+        props.input.onFocus && props.input.onFocus()
+    }
+    const onBlur = () => {
+        if (isSearchMode) {
+            // HACK: when clicking on a list item the onBlur event
+            // of the input happens first and the onClick event on the
+            // list item second. This timeout works around that problem
+            // but we probably need a more structural fix
+            setTimeout(() => {
+                setSearchMode(false)
+            }, 250)
```

Comment:
Yeah I agree, this seems fragile.

Reply:
`onMouseDown` happens before the inputs `onBlur`, so I switched to that in https://github.com/dhis2/reports-app/pull/249/commits/37e672bc25c7d7adceee8928f3cac2e93a09bb9e